File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        Stack<TreeNode> S = new Stack<>();
        List<Integer> res = new ArrayList<>();
        TreeNode T = root;
        while(T!=null) {
            S.push(T);
            T = T.left;
        }
        while(!S.isEmpty()){
            T = S.pop();
            res.add(T.val);
            T = T.right;
            while(T!=null) {
                S.push(T);
                T = T.left;
            }
        }
        return res;
    }
}

File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> inorderTraversal(TreeNode* root) {
        TreeNode* T = root;
        stack<TreeNode*> S;
        vector<int> res;
        while(T!=NULL){
            S.push(T);
            T = T->left;
        }
        while(!S.empty()){
            T = S.top();
            S.pop();
            res.push_back(T->val);
            T = T->right;
            while(T!=NULL){
                S.push(T);
                T = T->left;
            }
        }
        return res;

    }
};